server/middleware/query.go

```diff
@@ -0,0 +1,35 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/medibloc/panacea-oracle/panacea"
+	log "github.com/sirupsen/logrus"
+)
+
+type queryMiddleware struct {
+	panaceaQueryClient panacea.QueryClient
+}
+
+func NewQueryMiddleWare(queryClient panacea.QueryClient) *queryMiddleware {
+	return &queryMiddleware{
+		panaceaQueryClient: queryClient,
+	}
+}
+
+func (mw *queryMiddleware) Middleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		log.Debug("Retrieving the last block")
+		height, err := mw.panaceaQueryClient.GetLastBlockHeight(r.Context())
+		if err == nil {
+			log.Debugf("Set the previous value of the last block heignt. LastHeight: %v, SetHeight: %v", height, height-1)
+			r = r.WithContext(
+				panacea.SetQueryBlockHeightToContext(r.Context(), height-1),
```

Comment:
Could you explain why do you use the previous height of the latest block here?
Is it for (almost) immediate response?

Reply:
I think it is very rare case, but what if there is no data in the previous block? since the data is set in the latest block. 
In other words, it's the case that a transaction (to store data) is executed and somebody query the data at the same time. We don't have to care this edge case?

Reply:
Yes, that case can happen enough. 
But I think we should put more importance on quick response. 
I think a late response will cause more problems.

Reply:
Ah. I haven't thought about using the previous height of the latest block. Umm... it's trade-off as you and Hansol said..... Accuracy vs Latency..
Let me think about it..